tests/test_methods.py

```diff
@@ -103,7 +102,7 @@ def test_sgpr_qu():
 
     @tf.function
     def closure():
-        return - model.log_marginal_likelihood()
+        return -model.log_marginal_likelihood()
```

Comment:
FWIW, I find it more helpful to have the space there to highlight that it's the negative. Without the space it's easy to miss. (I actually couldn't find a reference to unary minus operator in PEP-8...)

Reply:
https://softwareengineering.stackexchange.com/a/189055, it is a general style rule for unary operators in many languages

Reply:
> No space should separate a unary operator and its operand except when the operator is a word such as typeof.

which is the case here. Also, clarity >> our own style conforming to other people's style (we're already deviating with upper-case one-character variable names, and for a good reason)

Reply:
> Avoid extraneous whitespace in the following situations: [...] Between a trailing comma and a following close parenthesis.
(ref: https://www.python.org/dev/peps/pep-0008/#pet-peeves)